### PR TITLE
Make vector size uint32_t(or size_t)

### PR DESCRIPTION
**Pull request by qarmin**
Part of https://github.com/godotengine/godot-proposals/issues/5618

C++ stl vector size returns `size_t`, but Godot returns `int`, so this PR changes return size from int to `vec_size`(typedef to `uint32_t`, but can be easily changed to `size_t` of other unsigned type)

Most of changes I done via simple regex and this PR is unfinished but I can finish it if its direction is correct.

This will help to use better types in entire Godot codebase by using proper types for certain tasks(vector size cannot be negative, so `int` is not too good type)

### CODE REVIEW

**Comment by bruvzg** [comment]
Since GDScript `int` is 64-bit and GDExtension convert all to `int64_t`, it might be better to use 64-bit int (or size_t) here as well (+ there's no reason to limit vector size to 4 GB).

**Comment by fire** [comment]
CHEERS! I've been waiting for this for years.

**Comment by qarmin** [comment]
To have exactly same results as before change such workaround could be used
```
Error CowData<T>::resize(int p_size) {
	ERR_FAIL_COND_V(p_size < 0, ERR_INVALID_PARAMETER);
```
would become
```
Error CowData<T>::resize(uint32_t p_size) {
	ERR_FAIL_COND_V(int(p_size) < 0, ERR_INVALID_PARAMETER);
```

File: core/templates/vector.h [hunk 1]
```diff
@@ -84,13 +84,13 @@ class Vector {
...
+	_FORCE_INLINE_ T get(vec_size p_index) { return _cowdata.get(p_index); }
+	_FORCE_INLINE_ const T &get(vec_size p_index) const { return _cowdata.get(p_index); }
+	_FORCE_INLINE_ void set(vec_size p_index, const T &p_elem) { _cowdata.set(p_index, p_elem); }
+	_FORCE_INLINE_ vec_size size() const { return _cowdata.size(); }
 	Error resize(int p_size) { return _cowdata.resize(p_size); }
 	Error resize_zeroed(int p_size) { return _cowdata.template resize<true>(p_size); }
-	_FORCE_INLINE_ const T &operator[](int p_index) const { return _cowdata.get(p_index); }
+	_FORCE_INLINE_ const T &operator[](vec_size p_index) const { return _cowdata.get(p_index); }
 	Error insert(int p_pos, T p_val) { return _cowdata.insert(p_pos, p_val); }
 	int find(const T &p_val, int p_from = 0) const { return _cowdata.find(p_val, p_from); }
```
Comment (AndreaCatania) [inline core/templates/vector.h:95]: This is WIP most likely, thought beware that `find` should return `uint32_t` too. Usually other libs define `INDEX_NONE = UINT32_MAX` that is possible to use to specify unknowns positions.

```
if( my_strings.find("TheText") == INDEX_NONE )
```

----
Note, `local_vector.h` returns `int64_t`, so return `-1` is not an undefined behavior; though from user prospective it's annoying convert `int64_t` to `uint32_t`, it usually requite an extra if condition - so I find much comfortable the `INDEX_NONE` solution.

**Review by AndreaCatania — COMMENTED** [review]
This is a really good change. Though I don't like the use of `vec_size` as it hides the type used. I would much like the direct use of `uint32_t`.

In any case, great job :)

**Comment by lawnjelly** [comment]
Just out of interest, does using a `typedef` offer advantages over using a template parameter, as is done in `LocalVector` (and several of the templates I think)?

e.g.
```
template <class T, class U = uint32_t, bool force_trivial = false>
class LocalVector {
```

(It could be that there is a reason to prevent non-standard Vector use, but it would be nice to clarify this.)

Does this mean gdscript etc would have to use named constants instead of checking for -1? (As `UINT32_MAX` is not the same as -1 in `int64_t`)

**Comment by lyuma** [comment]
might help #62585

I would not want uint32_t to be used directly, because that would make it difficulty to compile with 64-bit vector support (for supporting large files or lightmaps or other use cases)

**Pull request closed without merging**